src/plugins/analysis/device_tree/internal/schema.py

```diff
@@ -0,0 +1,172 @@
+                off_mem_rsvmap=int.from_bytes(
+                    binary[16:20],
+                    byteorder='big',
+                ),
+                version=int.from_bytes(
+                    binary[20:24],
+                    byteorder='big',
+                ),
+                last_comp_version=int.from_bytes(
+                    binary[24:28],
+                    byteorder='big',
+                ),
+                boot_cpuid_phys=int.from_bytes(
+                    binary[28:32],
+                    byteorder='big',
+                ),
+                size_dt_strings=int.from_bytes(
+                    binary[32:36],
+                    byteorder='big',
+                ),
+                size_dt_struct=int.from_bytes(
+                    binary[36:40],
+                    byteorder='big',
+                ),
+            )
+
+            if header.version > cls.MAX_VERSION:
+                raise IllegalHeaderError(f'Version may not exceed {cls.MAX_VERSION} but is {header.version}.')
+
+            dt_len = len(binary)
+            if header.totalsize > dt_len:
+                raise IllegalHeaderError(
+                    f'Value {header.totalsize} for totalsize is larger than the whole device tree.'
+                )
+            if header.size_dt_strings > dt_len:
+                raise IllegalHeaderError(
+                    f'Value {header.size_dt_strings} for size_dt_strings is larger than the whole device tree.'
+                )
+            if header.off_dt_strings > dt_len:
+                raise IllegalHeaderError(
+                    f'Value {header.off_dt_strings} for off_dt_strings is larger than the whole device tree.'
+                )
+            if header.size_dt_struct > dt_len:
+                raise IllegalHeaderError(
+                    f'Value {header.size_dt_struct} for size_dt_struct is larger than the whole device tree.'
+                )
+            if header.off_dt_struct > dt_len:
+                raise IllegalHeaderError(
+                    f'Value {header.off_dt_struct} for off_dt_struct is larger than the whole device tree.'
+                )
+
+            return header
+
+    offset: int = Field(
+        description='The offset where the device tree is located in the file.',
+    )
+    header: Header = Field(
+        description=(
+            'The struct as described in '
+            'https://devicetree-specification.readthedocs.io/en/stable/flattened-format.html#header '
+            'except it is missing the magic field.'
+        ),
+    )
+    string: str = Field(
+        description='The whole device tree in string format.',
+    )
+    model: Optional[str] = Field(
+        description=(
+            'The model as described in the spec.\n'
+            'https://devicetree-specification.readthedocs.io/en/latest/chapter2-devicetree-basics.html?highlight=model#model'
+        ),
+    )
+    description: Optional[str] = Field(
+        description='',
+    )
```

Comment:
Regarding the docs for this. Where is this defined? I could not find it in the spec.

Reply:
@jstucke Any ideas?

Reply:
I don't know if there is a spec for this. All I know is that this field frequently appears in device trees (and also it usually contains some interesting information)

Reply:
Interesting. I'll write that in the documentation.

Interestingly, this field does not show up in any of the device trees shipped with the Linux kernel (`cat $(fd -e dts) | rg description`).

Reply:
I looked through some device trees in my local DB and it seems the "description" field is present if OpenWRT is used (so maybe it's just an OpenWRT thing?)